### PR TITLE
Fix issue where collection deletion success message was not correct

### LINKED ISSUE #10813: Collection title is not interpolated into success_message
When I delete a collection, the "success message" does not have the name of the collection in the message. Instead it has the string with the python formatting markup still in it. I found this in a project running 5.1.1 but have not verified it on a new project or a new project running off the current main branch. 

Screenshot 2023-08-22 at 11 52 39 AM

For folks looking for a good first issue, the first step should be to verify this problem in a fresh dev setup.

### PR DESCRIPTION
**Pull request by lb-**
-   Fixes #10813
-   [X] Do the tests still pass?[^1]
-   [X] Does the code comply with the style guide?
-   [X] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [X] For new features: Has the documentation been updated accordingly? **N/A**

### Before

Screenshot 2023-08-24 at 6 54 35 am

### After

Screenshot 2023-08-24 at 7 38 45 am

### CODE REVIEW

File: wagtail/admin/views/collections.py [hunk 1]
```diff
@@ -182,6 +182,6 @@ def post(self, request, pk):
             # collection is non-empty; refuse to delete it
             return HttpResponseForbidden()
 
+        messages.success(request, self.get_success_message())
```
Comment (zerolab) [inline wagtail/admin/views/collections.py:185]: note:
`get_success_message` is defined on the `DeleteView` and does the necessary interpolation
https://github.com/wagtail/wagtail/blob/main/wagtail/admin/views/generic/models.py#L693-L696

**Review by zerolab — APPROVED** [review]
Nice one @lb-

File: wagtail/admin/views/collections.py (hunk 1 above)
Comment (lb-) [reply wagtail/admin/views/collections.py:185]: Yeah I think this code probably did not use the shared views historically so it has a mix of code that uses it and ... a mix of codes that does not :)

**Comment by lb-** [comment]
Thanks, I will merge this in shortly.

**Pull request merged**